sources/web/datalab/static/notebook-list.js

```diff
@@ -148,23 +148,62 @@ function initializeNotebookList(ipy, notebookList, newNotebook, events, dialog,
     }
   });
 
-  function checkVersion(versionInfo) {
-    if (versionInfo == undefined) {
-      return;
+  // Compare two arrays, each representing a semantic version in the form of [0,1,2]
+  // Returns -1, 0, 1. Returns null if they're malformed
+  function semverCompare(v1, v2) {
```

Comment:
Is semver a common word? Why not call it SemanticVersionCompare or VersionCompare or etc.